analytics_data_api/v0/models.py

```diff
@@ -220,3 +220,56 @@ class Meta:
     def get_course_user(cls, course_id, username):
         return cls.search().query('term', course_id=course_id).query(
             'term', username=username).execute()
+
+    @classmethod
+    def get_users_in_course(
+            cls,
+            course_id,
+            segments=None,
+            ignore_segments=None,
+            cohort=None,
+            enrollment_mode=None,
+            text_search=None,
+            order_by=None,
+            sort_order=None
+    ):
+        """
+        Construct a search query for all users in `course_id` and return
+        the Search object.  Raises `ValueError` if both `segments` and
+        `ignore_segments` are provided.
+        """
+        if segments and ignore_segments:
+            raise ValueError('Can not combine `segments` and `ignore_segments` parameters.')
+
+        order_by = order_by or 'username'
+        sort_order = sort_order or 'asc'
+
+        search = cls.search()
+        search.query = Q('bool', must=[Q('term', course_id=course_id)])
+
+        # Filtering/Search
+        if segments:
+            search.query.must.append(Q('bool', should=[Q('term', segments=segment) for segment in segments]))
+        if ignore_segments:
```

Comment:
I guess this could be an elif because segments and ignore_segments are mutually exclusive.  That's definitely just a suggestion, and I thought I'd just mention it.